apps/system/utils/kdbg_heapinfo.c

```diff
@@ -81,20 +98,45 @@ static void kdbg_heapinfo_group_threadlist(void)
 }
 #endif
 
+#if CONFIG_MM_NHEAPS > 1
+static void kdbg_init_total_info(void)
+{
+	total_info.total_heap_size = 0;
+	total_info.cur_free = 0;
+	total_info.largest_free_size = 0;
+	total_info.cur_dead_thread = 0;
+	total_info.sum_of_stacks = 0;
+	total_info.sum_of_heaps = 0;
+}
+#endif
+
 int kdbg_heapinfo(int argc, char **args)
 {
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 	int option;
-	bool show_group = false;
+	int display_flag = HEAPINFO_DISPLAY_ALL;
 	int mode = HEAPINFO_SIMPLE;
 	int pid = HEAPINFO_PID_ALL;
+#if CONFIG_MM_REGIONS > 1
+	int region_idx;
+#endif
+#if CONFIG_MM_NHEAPS > 1
```

Comment:
Please eliminate repeated if conditions.

Reply:
which if condition do you mean?